packages/flutter/lib/src/widgets/scroll_physics.dart

```diff
@@ -529,15 +533,23 @@ class RangeMaintainingScrollPhysics extends ScrollPhysics {
       enforceBoundary = false;
     }
     if (maintainOverscroll) {
-      // Force the new position to be no more out of range
-      // than it was before, if it was overscrolled.
-      if (oldPosition.pixels < oldPosition.minScrollExtent) {
-        final double oldDelta = oldPosition.minScrollExtent - oldPosition.pixels;
-        return newPosition.minScrollExtent - oldDelta;
-      }
-      if (oldPosition.pixels > oldPosition.maxScrollExtent) {
-        final double oldDelta = oldPosition.pixels - oldPosition.maxScrollExtent;
-        return newPosition.maxScrollExtent + oldDelta;
+      // Force the new position to be no more out of range than it was before, if:
+      //  * it was overscrolled
+      //  * and the position has shrunk, meaning that some content was removed. The
+      //    reason for this condition is that when new content is added, keeping
+      //    the same overscroll would mean that instead of showing it to the user,
+      //    all of it is being skipped by jumping to right to the max extent.
+      final bool shrunk = newPosition.minScrollExtent > oldPosition.minScrollExtent ||
+                          newPosition.maxScrollExtent < oldPosition.maxScrollExtent;
```

Comment:
we can probably ignore it for now but there's an interesting question about what to do in the case where the minScrollExtent shrinks but the maxScrollExtent grows, and so on. It also depends on whether the position is currently before the minScrollExtent or after the maxScrollExtent.

Reply:
Sure. Just wanted to make it more semantically clear, but turns out it's maybe not